source/common/http/utility.cc

```diff
@@ -387,6 +387,18 @@ void Utility::appendVia(RequestOrResponseHeaderMap& headers, const std::string&
   headers.appendVia(via, ", ");
 }
 
+void Utility::updateAuthority(RequestHeaderMap& headers, absl::string_view hostname,
+                              const bool append_xfh) {
+  if (hostname.empty()) {
+    return;
+  }
```

Comment:
That makes sense! We can retain the original behavior by removing the check. I do not think we can ASSERT() because host_rewrite_path_regex_->replaceAll in source/common/router/config_impl.cc could potentially return an empty string.

Reply:
I see. Alright then. Could you please extend the unit test to include the case of empty hostname?